File: codegen/results/v002/gemma4_26b_nothink/20260411_180639/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def run_tests():
    """
    Runs test cases for the calculate_discount function.
    Includes boundary conditions for quantities 9, 10, 49, and 50.
    """
    # List of test cases: (price, quantity, expected_output, description)
    test_cases = [
        (10.0, 9, 90.0, "Boundary: quantity 9 (No discount)"),
        (10.0, 10, 90.0, "Boundary: quantity 10 (10% discount)"),
        (10.0, 49, 441.0, "Boundary: quantity 49 (10% discount)"),
        (10.0, 50, 375.0, "Boundary: quantity 50 (25% discount)"),
        (100.0, 100, 7500.0, "General: Large quantity (25% discount)"),
        (5.0, 1, 5.0, "General: Single item (No discount)")
    ]

    for price, quantity, expected, description in test_cases:
        # Calculate actual result
        actual = calculate_discount(price, quantity)
        
        # Using round to handle potential floating point precision issues in comparison
        # We round to 2 decimal places as is standard for currency
        if round(actual, 2) == round(expected, 2):
            print(f"PASS: {description}")
        else:
            print(f"FAIL: {description} (expected: {expected}, got: {actual})")


if __name__ == "__main__":
    run_tests()

File: codegen/results/v002/gemma4_26b_nothink/20260411_180639/typescript/005_unit_test_writer/solution.ts
/**
 * Calculates the bulk discount based on quantity.
 * 
 * Logic:
 * - quantity < 10: no discount (100% of price)
 * - 10 <= quantity < 50: 10% discount (90% of price)
 * - quantity >= 50: 25% discount (75% of price)
 * 
 * @param price - The unit price of the item
 * @param quantity - The number of items being purchased
 * @returns The total price after applying discounts
 */
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

/**
 * Represents a single test case
 */
interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

/**
 * Executes the test suite and prints results to stdout
 */
function runTests() {
    const testCases: TestCase[] = [
        {
            description: "Boundary: quantity 9 (No discount)",
            price: 100,
            quantity: 9,
            expected: 900
        },
        {
            description: "Boundary: quantity 10 (10% discount applied)",
            price: 100,
            quantity: 10,
            expected: 900 // 10 * 100 * 0.9
        },
        {
            description: "Boundary: quantity 49 (10% discount applied)",
            price: 100,
            quantity: 49,
            expected: 4410 // 49 * 100 * 0.9
        },
        {
            description: "Boundary: quantity 50 (25% discount applied)",
            price: 100,
            quantity: 50,
            expected: 3750 // 50 * 100 * 0.75
        },
        {
            description: "Standard case: quantity 1 (No discount)",
            price: 50,
            quantity: 1,
            expected: 50
        },
        {
            description: "Large quantity: quantity 100 (25% discount applied)",
            price: 10,
            quantity: 100,
            expected: 750 // 100 * 10 * 0.75
        }
    ];

    let passedCount = 0;
    let failedCount = 0;

    for (const test of testCases) {
        const actual = calculateDiscount(test.price, test.quantity);
        
        // Using a small epsilon for floating point comparison
        const isCorrect = Math.abs(actual - test.expected) < 0.000001;

        if (isCorrect) {
            console.log(`PASS: ${test.description}`);
            passedCount++;
        } else {
            console.log(`FAIL: ${test.description} (expected: ${test.expected}, got: ${actual})`);
            failedCount++;
        }
    }

    console.log(`\nTest Summary: ${passedCount} passed, ${failedCount} failed.`);
}

// Execute the program
runTests();